### DISCUSSION TITLE
Retrieving timeline variables across trials

### DISCUSSION THREAD

**Discussion by wsr017** [jspsych/jsPsych#839] [Q&A] [answered]
Hello all,

Thanks so much for participating in this forum! I'm new to jsPsych and feel like I'm missing something simple here. I have an rdk trial followed by a forced direction choice trial, where the subject either presses the right or left arrow to indicate the coherent direction of the dots. The correct keyboard response is stored using the timeline variable `correct_key: 39` or `correct_key: 37`.

```
var dot_motion = {
    type: "rdk",
    text: "",
    number_of_dots: n_dots,
    coherence: jsPsych.timelineVariable('coher_level'),
    move_distance: dot_speed, 
    reinsertType: 1, 
    aperture_type: 3, 
    RDK_type: 3, 
    background_color: '#d2d2d2',
    dot_color: 'black',
    aperture_width: dot_aperture_width,
    aperture_height: dot_aperture_width,
    aperture_center_y: dot_aperture_center_y,
    aperture_center_x: dot_aperture_center_x,
    response_ends_trial: true,
    trial_duration: dotsInterval,
    response_ends_trial: true,
    coherent_direction: jsPsych.timelineVariable('dotDir'),
    oppositeCoherence: 0,
    dot_life: dot_lifetime,
    dot_radius: dot_radius,
    data: {congruence: jsPsych.timelineVariable('cong'), correct_key: jsPsych.timelineVariable('correct_key')}

  }

  var direction_choice = {
    type: 'html-keyboard-response',
    stimulus: '<div style="text-align: center;width:600px;margin: auto;"><p><b><u>Dots were moving left or right? (Use the arrow keys) </u></b></p></div>',
    choices: [37, 39],
    on_finish: function(data){
      direction_pressed = data.key_press;
      correct_answer = jsPsych.timelineVariable('correct_key'); 
      right_or_wrong = (direction_pressed == correct_answer);
      if(right_or_wrong){
        data.correct = true;
      } else {
        data.correct = false;
      }
    },
  }
```
I am trying to label the responses as correct or incorrect, but right_or_wrong always evaluates to false because the value of correct_answer is "function () { return timeline.timelineVariable(varname); }" when I look at it in the console. direction_pressed seems to work correctly. 

Is there something I'm missing in evaluating correct_answer? Thanks so much for the help! 

(I apologize for the confusing variable names!)

**Comment by jodeleeuw** [comment, marked as answer]
Hi @wsr017,

When you use `jsPsych.timelineVariable()` inside a function you have to set the second parameter to `true`. 

```js
correct_answer = jsPsych.timelineVariable('correct_key', true); 
```

The docs describe why here: https://www.jspsych.org/overview/timeline/#timeline-variables. This is a common mistake that we're thinking about how to make more intuitive! (see #788)

**Reply by wsr017** [reply]
Thank you so much!! Sorry for repeating an old post!
